src/services/lobby.service.ts

```diff
@@ -26,36 +29,45 @@ class LobbyService
   async joinLobby(userId: string, gameId: string): Promise<IGame> 
   {
     const game = await gameRepository.getById(gameId);
+
     if (!game) 
     {
       throw new HttpException(HttpStatusCode.NOT_FOUND, 'Lobby not found.');
     }
   
-    const isUserInLobby = game.teams.some(team => team.members.includes(userId));
-    if (isUserInLobby) {
+    const isUserInLobby = Array.from(this.teamMembers.values()).some(members => members.has(userId));
+
+    if (isUserInLobby) 
+    {
       throw new HttpException(HttpStatusCode.BAD_REQUEST, 'User already in the lobby.');
     }
 
     let assigned = false;
-    for (const team of game.teams) {
-      if (team.members.length < game.options.maxPlayersPerTeam) {
-        team.members.push(userId);
+
+    for (const team of game.teams) 
+    {
+      const members = this.teamMembers.get(team.teamId) ?? new Set<string>();
+      if (members.size < game.options.maxPlayersPerTeam) 
+      {
+        members.add(userId);
+        this.teamMembers.set(team.teamId, members);
         assigned = true;
         break;
       }
     }
     
-    if (!assigned) {
-      if (game.teams.length < MAX_TEAMS) {
-        const newTeam = {
-          teamId: `team-${game.teams.length + 1}`,
-          score: 0,
-          members: [userId]
-        };
-        game.teams.push(newTeam);
-      } else {
-        throw new HttpException(HttpStatusCode.BAD_REQUEST, 'All teams are full, and no more teams can be created.');
-      }
+    if (!assigned) 
+    {
+      const newTeamId = `team-${game.teams.length + 1}`;
+      game.teams.push({
```

Comment:
it wont work because `game.teams` stores ids of teams from couchdb, but here you put some random string, so all logic from real game service wont work

Reply:
On last meeting we discussed user flow, so basically, client creates account, next they create team or are added to the existing team, after it game can be started with two or more teams